nx5 io：Fixed the problem of abnormal default status of pwm-fan and abnormal status of sound card

vamrs-feng opened the pull request:
1. The incorrect sound card name prevents the pluseaudio service from 
properly obtaining sound card related information;
2.  cherry pick form mainline 11be938

vamrs-feng committed:
arm64: dts: nx5 io: Modify dp sound card name

Signed-off-by: Feng Zhang <feng@radxa.com>

RadxaYuntian commented on arch/arm64/boot/dts/rockchip/rk3588s-radxa-nx5-io.dts:
```diff
@@ -90,7 +90,9 @@
 &pwm12 {
-	pinctrl-0 = <&pwm12m1_pins>;
+	pinctrl-names = "default", "active";
+	pinctrl-0 = <&pwm12m1_pins_default>;
```
pwm12m1_pins不是4 RK_PB5么？看你的意思nx5的风扇是接在非硬件pwm上吧？这样要用gpio-pwm先虚拟出一个pwm设备，再把风扇配置到这个pwm控制器上：https://github.com/radxa/overlays/blob/main/arch/arm64/boot/dts/amlogic/overlays/radxa-zero2-radxa-25w-poe.dts#L28-L41

RadxaYuntian commented on drivers/pwm/pwm-rockchip.c:
```diff
@@ -251,6 +251,8 @@ static int rockchip_pwm_apply(struct pwm_chip *chip, struct pwm_device *pwm,
 	if (state->enabled)
 		ret = pinctrl_select_state(pc->pinctrl, pc->active_state);
+	else
+		pinctrl_select_default_state(chip->dev);
```
这是linux内核的bug，我觉得没必要管。但如果要修复则可以用这个补丁：https://github.com/torvalds/linux/commit/a2ca752055edd39be38b887e264d3de7ca2bc1bb

RadxaYuntian commented:
建议采用review里面说的那个上游补丁。你这个改动影响所有的pwm设备

vamrs-feng committed:
pwm: rockchip: Enable clock before calling clk_get_rate()

The documentation for clk_get_rate() in include/linux/clk.h states the
function's result is valid only for a clock source that has been
enabled. However, the Rockchip PWM driver uses this function in two places
to query the rate of a clock without first ensuring it is enabled.

Fix this by modifying rockchip_pwm_get_state() and rockchip_pwm_apply() so
they enable a device's PWM clock before querying its rate (in the latter
case, the querying is actually done in rockchip_pwm_config()) and disable
the clock again before returning.

Reported-by: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
Signed-off-by: Simon South <simon@simonsouth.net>
Reviewed-by: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
Signed-off-by: Thierry Reding <thierry.reding@gmail.com>

RadxaYuntian commented on drivers/pwm/pwm-rockchip.c:
现在没有复现了么
